### PR TITLE
Saspfeatures client

### PR DESCRIPTION
**Pull request by yashmohod**

### CODE REVIEW

File: src/Components/SaspReferal/SaspReferal.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+import InputGroup from 'react-bootstrap/InputGroup';
+
+
+export default function SaspReferal(props) {
+
+ 
+    useEffect(() => 
+    {   
+
+    },[])
```
Comment (pichan321) [inline src/Components/SaspReferal/SaspReferal.js:13]: this useEffect should be removed if not used to do anything, also remove useEffect from import too.

File: src/Components/TimePicker24H/TimePicker24H.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+import { useState , React, useEffect} from "react";
+import { Form} from 'react-bootstrap';
+import InputGroup from 'react-bootstrap/InputGroup';
+
+
+export default function TimePicker24H(props) {
+
+    // const [time, setTime] = useState("");
+    const[timeH,settimeH] = useState("00");
+    const[timeM,settimeM] = useState("00");
```
Comment (pichan321) [inline src/Components/TimePicker24H/TimePicker24H.js:11]: There are a bunch of time picker libraries out there for you to corporate. I wouldn't mind using this though.

File: src/Pages/Dashboard/Dashboard.js [hunk 3]
```diff
@@ -1,12 +1,27 @@
...
-import { Link } from "react-router-dom";
 import './Dashboard.css'
-import { API_URL, post } from "../../Utils/API";
+import { API_URL, post,get } from "../../Utils/API";
+import { ToastContainer, toast } from 'react-toastify';
+import {Features} from "./features"
 
 export default function({loggedIn, setLoggedIn, loggedInUser}) {
     const [clockin, setClockin] = useState(false)
+    const [org,setorg] = useState("")
```
Comment (pichan321) [inline src/Pages/Dashboard/Dashboard.js:12]: setorg should be setOrg for readability

Comment (pichan321) [reply src/Pages/Dashboard/Dashboard.js:12]: setpos should also be setPos

File: src/Pages/Dashboard/Dashboard.js [hunk 4]
```diff
@@ -28,52 +53,66 @@ export default function({loggedIn, setLoggedIn, loggedInUser}) {
...
-     {title: "Incidents", description:"Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.", url: "/incidents"},
-     {title: "Locations", description:"Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.", url: "/locations"},
-     {title: "Employee Accounts", description:"Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.", url: "/employee-accounts"},
-     {title: "Register Accounts", description:"Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.", url: "/register-accounts"},
-     {title: "Change Passwords", description:"Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.", url: "/change-passwords"},
+    [{org:"SASP",accessLevel:0,title: "Daily", description:"", url: "/SASPpages/daily", external_url:''},
```
Comment (pichan321) [inline src/Pages/Dashboard/Dashboard.js:60]: if there's no link for external_url, you don't need to add the property to the object. You can remove it.

File: src/Pages/EmployeeAccounts/EmployeeAccounts.js [hunk 5]
```diff
@@ -0,0 +1,346 @@
...
+
+  
+
+  
+
+    async function getAccounts() {
+        let response = await get(API_URL + "/getAllAccounts?token=" +  localStorage.getItem("token"))
+        response = JSON.parse(response.accounts)
+
+        for(let x =0; x< response.length; x++){
```
Comment (pichan321) [inline src/Pages/EmployeeAccounts/EmployeeAccounts.js:235]: should use forEach loop here

File: src/Pages/SASPpages/Daily/Daily.js [hunk 6]
```diff
@@ -0,0 +1,267 @@
...
+  <Nav setLoggedIn={setLoggedIn} loggedInUser={loggedInUser}/>
+  const navigate = useNavigate();
+  const [referals,setreferals]=useState([]);
+  const [referalData, setreferalData] = useState([]);
+  const [referalsCount,setreferalsCount] = useState(0);
+  const [showReferals,setshowReferals]=useState(false);
+
+
+
+useEffect(() =>
```
Comment (pichan321) [inline src/Pages/SASPpages/Daily/Daily.js:25]: this useEffect is doing nothing. Can we remove this?

**Review by pichan321 — CHANGES_REQUESTED** [review]

File: src/Components/TimePicker24H/TimePicker24H.js (hunk 2 above)
Comment (yashmohod) [reply src/Components/TimePicker24H/TimePicker24H.js:11]: I tried to look up date pickers libraries but none used the 24h format. 
If you can find one with 24h format we can incorporate it into our project.

File: src/Pages/Dashboard/Dashboard.js (hunk 3 above)
Comment (yashmohod) [reply src/Pages/Dashboard/Dashboard.js:12]: agreed and I have made the changes

File: src/Pages/Dashboard/Dashboard.js (hunk 4 above)
Comment (yashmohod) [reply src/Pages/Dashboard/Dashboard.js:60]: oh, I see ok then i would remove them

**Pull request merged**